File: backend/app/schemas/brand_kit.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class BrandKitVersionInput(BaseModel):
    brand_name: str = Field(min_length=1, max_length=200)
    positioning: str = Field(min_length=1, max_length=4000)
    default_language: str = Field(min_length=1, max_length=100)
    brand_tone: str = Field(min_length=1, max_length=2000)
    preferred_terms: list[str] = Field(default_factory=list)
    forbidden_terms: list[str] = Field(default_factory=list)
    target_regions: list[str] = Field(default_factory=list)
    audience_guidelines: list[str] = Field(default_factory=list)
    visual_guidelines: list[str] = Field(default_factory=list)
    required_disclosures: list[str] = Field(default_factory=list)
    claims_constraints: list[str] = Field(default_factory=list)


class BrandKitCreate(BaseModel):
    name: str = Field(min_length=1, max_length=200)
    version: BrandKitVersionInput


class BrandKitVersionRead(BrandKitVersionInput):
    model_config = ConfigDict(from_attributes=True)

    id: int
    brand_kit_id: int
    version_number: int
    digest: str
    created_at: datetime


class BrandKitRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    created_at: datetime
    updated_at: datetime
    versions: list[BrandKitVersionRead] = Field(default_factory=list)


class BrandKitVersionCreateRead(BaseModel):
    version: BrandKitVersionRead
    reused: bool


class ProductBrandKitBinding(BaseModel):
    brand_kit_id: int = Field(gt=0)
    brand_kit_version_id: int = Field(gt=0)

File: frontend/src/types/brandKit.ts
export interface BrandKitVersionInput {
  brand_name: string;
  positioning: string;
  default_language: string;
  brand_tone: string;
  preferred_terms: string[];
  forbidden_terms: string[];
  target_regions: string[];
  audience_guidelines: string[];
  visual_guidelines: string[];
  required_disclosures: string[];
  claims_constraints: string[];
}

export interface BrandKitVersion extends BrandKitVersionInput {
  id: number;
  brand_kit_id: number;
  version_number: number;
  digest: string;
  created_at: string;
}

export interface BrandKit {
  id: number;
  name: string;
  created_at: string;
  updated_at: string;
  versions: BrandKitVersion[];
}

export interface BrandKitCreatePayload {
  name: string;
  version: BrandKitVersionInput;
}

export interface BrandKitVersionCreateResult {
  version: BrandKitVersion;
  reused: boolean;
}

export interface ProductBrandKitBindingPayload {
  brand_kit_id: number;
  brand_kit_version_id: number;
}
